tests/unit/s2n_server_psk_extension_test.c

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#include "crypto/s2n_hmac.h"
+#include "tls/extensions/s2n_server_psk.h"
+#include "tls/s2n_tls.h"
+#include "utils/s2n_bitmap.h"
+
+#define TEST_PSK_WIRE_INDEX 1
+#define TEST_PSK_HMAC S2N_HMAC_SHA384
+
+uint8_t test_identity[] = "test identity";
+uint8_t test_secret[] = "test secret";
+
+static s2n_result validate_psk_is_wiped(struct s2n_psk *psk)
+{
+    ENSURE_EQ(psk->identity.data, NULL);
+    ENSURE_EQ(psk->identity.size, 0);
+    ENSURE_EQ(psk->secret.data, NULL);
+    ENSURE_EQ(psk->secret.size, 0);
+    ENSURE_EQ(psk->early_secret.data, NULL);
+    ENSURE_EQ(psk->early_secret.size, 0);
+
+    return S2N_RESULT_OK;
+}
+
+static s2n_result validate_psk_is_not_wiped(struct s2n_psk *psk)
+{
+    ENSURE_REF(psk->identity.data);
+    ENSURE_NE(psk->identity.size, 0);
+    ENSURE_REF(psk->secret.data);
+    ENSURE_NE(psk->secret.size, 0);
+
+    return S2N_RESULT_OK;
+}
+
+
+static s2n_result setup_client_psks(struct s2n_connection *client_conn)
+{
+    ENSURE_REF(client_conn);
+
+    struct s2n_psk *shared_psk = NULL;
```

Comment:
Nit: shared_psk should probably be declared down on line 67, like other_client_psk is.